code/datums/brain_damage/creepy_trauma.dm

```diff
@@ -0,0 +1,117 @@
+/datum/brain_trauma/special/creep
+	name = "Erotomania"
+	desc = "Patient has a subtype of delusional disorder, becoming irrationally attached to someone."
+	scan_desc = "severe erotomaniac delusions"
+	gain_text = ""
+	lose_text = "<span class='warning'>You no longer feel so attached.</span>"
+	can_gain = TRUE
+	resilience = TRAUMA_RESILIENCE_SURGERY
+	var/mob/living/obsession
+	var/datum/objective/spendtime/attachedcreepobj
+	var/datum/antagonist/creep/antagonist
+
+	var/total_time_creeping = 0 //just for roundend fun
+	var/time_spent_away = 0
+	var/obsession_hug_count = 0
+
+/datum/brain_trauma/special/creep/on_gain()
+
+	//setup, linking, etc//
+	..()
+	if(!obsession)//admins didn't set one
+		obsession = find_obsession()
+		if(!obsession)//we didn't find one
+			qdel(src)
+	owner.mind.add_antag_datum(/datum/antagonist/creep)
+	antagonist = owner.mind.has_antag_datum(/datum/antagonist/creep)
+	antagonist.trauma = src
+	//antag stuff, //
+	antagonist.forge_objectives(obsession.mind)
+	antagonist.greet()
+
+
+
+/datum/brain_trauma/special/creep/on_life()
+	if(!obsession)//gibbing, etc
+		return
+	if(obsession.stat == DEAD) //killing them "cures" you! kinda! ish!
+		return
+	var/foundyou = FALSE
+	for(var/mob/living/L in range(7, owner))
```

Comment:
really? just do a get_dist between get_turf(self) and the get_turf(obsession) first so you only do the view check if they're actually within 7 tiles.

 there's no reason to look at every thing in a 7 block range every time just to find one mob you already have a reference to to do a distance check

Reply:
i'll do that, it's much better

Reply:
And while i'm here I should really change that to view instead of range.